src/experiments.js

```diff
@@ -56,6 +64,128 @@ export function isCanary(win) {
 
 
 /**
+ * Enable experiments detailed in an origin trials token iff the token is
+ * valid. A token is invalid if *
+ *   1. The token is malformed (e.g. non-existant version number)
+ *   2. The token is not for this origin
+ *   3. The experiments data was not signed with our private key
+ * @param {!Window} win
+ * @param {string} token
+ * @param {!Object=} opt_publicJwk Used for testing only.
+ * @return {!Promise}
+ */
+function enableExperimentsFromToken(win, token, opt_publicJwk) {
+  const crypto = cryptoFor(win);
+  if (!crypto.isCryptoAvailable()) {
+    user().warn(TAG, 'Crypto is unavailable');
+    return Promise.resolve();
+  }
+  /**
+   * token = encode64(version + length + config + sign(config, private_key))
```

Comment:
Continuing from: https://github.com/ampproject/amphtml/pull/8944#discussion_r115806286

Should be `sign(version + length + config, private_key)`.